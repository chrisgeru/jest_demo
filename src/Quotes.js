import _ from "lodash";

export const getRandomQuote = () => {
    const index = _.random(0, 18);
    return fetch(`https://my-json-server.typicode.com/chrisgeru/quotes/posts/${index}`)
        .then(response => response.json())
        .then(json => json.title);
};
