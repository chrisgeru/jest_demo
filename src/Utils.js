export function resolveQuotes() {
    return new Promise((resolver) => setTimeout(() => resolver()))
}
