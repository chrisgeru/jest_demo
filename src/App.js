import React, {Component} from 'react';
import './App.css';
import {getRandomQuote} from "./Quotes";

export class Home extends React.Component {

    render() {
        return (
            <h1 className="App">
                {this.props.quote}
            </h1>
        )
    }
}

export class App extends Component {

    state = {quote: ""};

    async componentDidMount() {
        const quote = await getRandomQuote();
        this.setState({quote})
    }

    render() {
        return (
            <Home quote={this.state.quote}/>
        );
    }
}
