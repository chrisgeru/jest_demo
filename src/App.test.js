import React from 'react';
import {configure, mount, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import {getRandomQuote} from './Quotes';

import {App, Home} from "./App"
import {resolveQuotes} from "./Utils";

jest.mock('./Quotes');

configure({adapter: new Adapter()});

it('renders as expected', () => {
    const wrapper = shallow(<Home quote={"Hello World"}/>);
    expect(wrapper).toMatchSnapshot();

});

it('works as expected', async () => {
    getRandomQuote.mockResolvedValueOnce('Readability counts.');
    const wrapper = mount(<App />);

    await resolveQuotes();
    wrapper.update();
    expect(wrapper.find('h1').text()).toBe('Readability counts.')
});
