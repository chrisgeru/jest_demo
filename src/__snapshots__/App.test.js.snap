// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<h1
  className="App"
>
  Hello World
</h1>
`;

exports[`works as expected 1`] = `
<App>
  <Home
    quote="Test"
  >
    <h1
      className="App"
    >
      Test
    </h1>
  </Home>
</App>
`;
